common/check_internal.h

```diff
@@ -64,9 +93,10 @@ template <TemplateString Kind, TemplateString File, int Line,
     // at all, just the condition.
     CheckFailImpl(Kind.c_str(), File.c_str(), Line, ConditionStr.c_str(), "");
   } else {
-    CheckFailImpl(
-        Kind.c_str(), File.c_str(), Line, ConditionStr.c_str(),
-        llvm::formatv(FormatStr.c_str(), std::forward<Ts>(values)...).str());
+    CheckFailImpl(Kind.c_str(), File.c_str(), Line, ConditionStr.c_str(),
+                  llvm::formatv(FormatStr.c_str(),
+                                ConvertFormatValue(std::forward<Ts>(values))...)
```

Comment:
We might need this in more places, but we can worry about that later. This is an improvement for now.